testing/integration/switch_privileged_test.go

```diff
@@ -0,0 +1,139 @@
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSwitchPrivilegedWithoutBasePath(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		// We require sudo for this test to run
+		// `elastic-agent install`.
+		Sudo: true,
+
+		// It's not safe to run this test locally as it
+		// installs Elastic Agent.
+		Local: false,
+	})
+
+	// Get path to Elastic Agent executable
+	fixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+
+	// Prepare the Elastic Agent so the binary is extracted and ready to use.
+	err = fixture.Prepare(ctx)
+	require.NoError(t, err)
+
+	// Run `elastic-agent install`.  We use `--force` to prevent interactive
+	// execution.
+	opts := &atesting.InstallOpts{Force: true, Privileged: false}
+	out, err := fixture.Install(ctx, opts)
+	if err != nil {
+		t.Logf("install output: %s", out)
+		require.NoError(t, err)
+	}
+
+	// Check that Agent was installed in default base path in unprivileged mode
+	require.NoError(t, installtest.CheckSuccess(ctx, fixture, opts.BasePath, true))
+
+	// Switch to privileged mode
+	out, err = fixture.Exec(ctx, []string{"privileged", "-f"})
+	if err != nil {
+		t.Logf("privileged output: %s", out)
+		require.NoError(t, err)
+	}
+
+	// Check that Agent is running in default base path in privileged mode
+	require.NoError(t, installtest.CheckSuccess(ctx, fixture, opts.BasePath, false))
+}
+
+func TestSwitchPrivilegedWithBasePath(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		// We require sudo for this test to run
+		// `elastic-agent install`.
+		Sudo: true,
+
+		// It's not safe to run this test locally as it
+		// installs Elastic Agent.
+		Local: false,
+	})
+
+	// Get path to Elastic Agent executable
+	fixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+
+	// Prepare the Elastic Agent so the binary is extracted and ready to use.
+	err = fixture.Prepare(ctx)
+	require.NoError(t, err)
+
+	// When running in unprivileged using a base path the
+	// base needs to be accessible by the `elastic-agent-user` user that will be
+	// executing the process, but is not created yet. Using a base that exists
+	// and is known to be accessible by standard users, ensures this tests
+	// works correctly and will not hit a permission issue when spawning the
+	// elastic-agent service.
+	var basePath string
+	switch runtime.GOOS {
+	case define.Linux:
+		basePath = `/usr`
+	case define.Windows:
+		basePath = `C:\`
+	default:
+		// Set up random temporary directory to serve as base path for Elastic Agent
+		// installation.
+		tmpDir := t.TempDir()
+		basePath = filepath.Join(tmpDir, strings.ToLower(randStr(8)))
```

Comment:
Why do we need to add a randomizedString to `tmpDir` ? `tmpDir` will already contain the testcase name and a progressive number...

On the same topic: why don't we use TmpDir for all OSes ? Does it not work on windows/linux ?

Reply:
This comes from the install tests, there are reasons for using different directories. It has to do with the permissions on each OS. Possibly something that could be changed, but this PR is already large enough and not something I want to change here. If they are changed here then they should be changed in all the install tests as well.